### ISSUE TITLE
API: consider deprecating DataFrame.hist in favor of DataFrame.plot.hist

### ISSUE THREAD

**Issue by shoyer** [pydata/pandas#11053]
This came up in the PR adding plot submethods like `.plot.hist`:  https://github.com/pydata/pandas/pull/9321#issuecomment-139203762

As long as API differences can be reconciled, I think it would be a nice idea to deprecate (and eventually remove) `DataFrame.hist`, `DataFrame.boxplot` and `Series.hist` in favor of the accessor mtehods.

This old issue discusses API differences (possibly no longer accurate/relevant?): https://github.com/pydata/pandas/issues/413

**Comment by jreback** [comment]
@shoyer @TomAugspurger either one of you for 0.18.0?

**Comment by ResidentMario** [comment]
Excuse my naivete, but what is to be done here? I did a quick test and all of the [`DataFrame.plot`](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.plot.html) arguments seem to work when passed into sub-methods.

**Comment by jreback** [comment]
``DataFrame.hist``, ``DataFrame.boxplot`` and ``Series.hist `` in favor of

``DataFrame.plot.hist`` etc

this issue is about emitting a deprecation for these (the replace is already there)

**Comment by ResidentMario** [comment]
Is this as simple as tossing a bunch of deprecate decorators onto things, then?

**Comment by jreback** [comment]
@ResidentMario 

I think these *are* actually defined in ``pandas.tools.plotting.hist_series`` as a real method. I *think* that should be removed and instead direct to the plotting accessor.

@TomAugspurger @jorisvandenbossche

**Comment by ResidentMario** [comment]
Well, the `hist_series` code does include this comment:

    # hack until the plotting interface is a bit more unified

So it does indeed sound like things are a bit more complicated than that.

**Comment by jreback** [comment]
yeah, plotting is like Alice-in-wonderland..........

**Comment by jbrockmendel** [comment]
Out of curiosity, why are `plot`, `hist`, and `boxplot` defined at the [bottom](https://github.com/pandas-dev/pandas/blob/master/pandas/core/frame.py#L5974) of frames.py instead of included in the class definition?  [Same](https://github.com/pandas-dev/pandas/blob/master/pandas/core/series.py#L3067) goes for `Series.plot` and `Series.hist`.

**Comment by jreback** [comment]
@jbrockmendel I think these are all historical. No reason they should not be deprecated / fixed. (refering to.hist and .boxplot); .plot is by-definition the namespace

**Comment by jreback** [comment]
anyone want to do this deprecation?

**Comment by mangecoeur** [comment]
I've noticed that `.hist()` and `.plot.hist()` give different outputs (pandas 0.22, matplotlib 2.2) - I opened issue #21048 . I don't know if this happens in all cases but I guess this should be addressed before `hist()` is replaced by `plot.hist()`

**Comment by TomAugspurger** [comment]
Right, part of the deprecation would be providing a way to achieve the behavior of `DataFrame.hist` within `.plot.hist`. Copying over your example here.

---

from @mangecoeur in https://github.com/pandas-dev/pandas/issues/21048#issue-323152699

#### Code Sample, a copy-pastable example if possible

```python
cols = ['heat_total', 'area_density', 'mean_n_floors',
       'n_buildings', 'area_demand_density_grid', 'floor_demand_density']

plt.figure()
df[cols].plot.hist(
    bins=20,
    alpha=0.5
)

plt.figure()
df[cols].hist(
    bins=20,
    alpha=0.5
)

```
#### Problem description

`df.hist()` and `df.plot.hist()` give different plots: `.hist()` plots a grid of figures while `.plot.hist()` plots overlapping distributions.

Result of `df.hist()`

![hist](https://user-images.githubusercontent.com/743508/40050900-243630f4-5839-11e8-99cc-ff2f4421f8f6.png)

Result `df.plot.hist()`

![plothist](https://user-images.githubusercontent.com/743508/40050901-244fb114-5839-11e8-9676-846d8b9a90ba.png)


#### Expected Output

`.hist()` and `.plot.hist()` should give exactly the same results

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.3.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.22.0
pytest: 3.2.1
pip: 9.0.1
setuptools: 36.5.0.post20170921
Cython: 0.26.1
numpy: 1.14.2
scipy: 0.19.1
pyarrow: None
xarray: 0.10.0
IPython: 6.2.1
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: None
matplotlib: 2.2.2
openpyxl: 2.4.9
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.1.0
bs4: 4.6.0
html5lib: 0.999999999
sqlalchemy: 1.1.13
pymysql: None
psycopg2: 2.7.4 (dt dec pq3 ext lo64)
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>
